meteor/server/api/playout/playout.ts

```diff
@@ -212,17 +221,27 @@ export namespace ServerPlayoutAPI {
 			let pBlueprint = makePromise(() => getBlueprintOfRundown(rundown))
 
 			const currentPart = rundown.currentPartId ? rundownData.partsMap[rundown.currentPartId] : undefined
-			if (currentPart && currentPart.transitionDuration) {
+			if (currentPart) {
 				const prevPart = rundown.previousPartId ? rundownData.partsMap[rundown.previousPartId] : undefined
 				const allowTransition = prevPart && !prevPart.disableOutTransition
+				const start = currentPart.getLastStartedPlayback()
 
 				// If there was a transition from the previous Part, then ensure that has finished before another take is permitted
-				if (allowTransition) {
-					const start = currentPart.getLastStartedPlayback()
+				if (allowTransition && currentPart.transitionDuration) {
 					if (start && now < start + currentPart.transitionDuration) {
 						return ClientAPI.responseError('Cannot take during a transition')
 					}
 				}
+
+				const offset = currentPart.getLastPlayOffset()
+				if (start && offset && currentPart.expectedDuration) {
+					const t = Date.now() - start! + offset!
```

Comment:
Shouldn't this be `const t = Date.now() - (start + offset)`? Perhaps also rename `t` to something more descriptive, like `actualDuration` or the like?

Reply:
`Date.now()` and `start` are timestamps I believe. Offset is a duration. so `Date.now() - start` gives a duration, if you add the offset you get where in the part we are.